packages/runtime/container-runtime/src/dataStoreRegistry.ts

```diff
@@ -15,6 +15,13 @@ export class FluidDataStoreRegistry implements IFluidDataStoreRegistry {
 
     constructor(namedEntries: NamedFluidDataStoreRegistryEntries) {
         this.map = new Map(namedEntries);
```

Comment:
this will lead to multiple enumerations of the namedEntries as the map will do it internally, then you also do it below.

I would prefer if we started with an empty map, and added entries one by one, and checking if the names exists.

Reply:
@clarenceli-msft - You could also do `this.map.set()` and `count++` in the same loop, if you like your current micro-optimization.

Reply:
> @clarenceli-msft - You could also do `this.map.set()` and `count++` in the same loop, if you like your current micro-optimization.

I did not use the count variable right now. I just check if the new name exists in the current map or not, do you think its okay?